find_exponential/src/scoring.rs

```diff
@@ -0,0 +1,42 @@
+use ndarray::{Array1, Array2};
+use ndarray_stats::{CorrelationExt, SummaryStatisticsExt};
+use itertools::Itertools;
+
+#[allow(dead_code)]
+pub fn pearson_correlation(array: &Array2<f64>) -> (f64, bool) {
+    let corr = array.pearson_correlation()[[1, 0]];
+    (corr, corr < super::ACCEPTANCE_CORRELATION)
+}
+
+#[allow(dead_code)]
+pub fn slope_stddev(array: &Array2<f64>) -> (f64, bool) {
```

Comment:
Can you explain how this works?

Reply:
I'll add a comment.

Imagine the `(num_bytes, duration)`-pairs being an `(x, y)`-coordinate. This function calculates the slope between all possible pairs of coordinates, keeping only positive slopes (to keep A—B, but remove B­—A) (I should rewrite this as nested for-loop). Then I calculate the standard deviation of those slopes. In a perfect linear world, all slopes are equal and the standard deviation is zero. With non-linear growth, the standard deviation will be much higher as the slopes increase quadratically.

Taking into account all combinations of coordinates (A—B, A—C, B—C) instead of only adjacent ones (A—B, B—C) I reduce the impact of outliers as their impact will be smaller in the mean.

Reply:
Right — that is a sensible start I think. Though we may need to look at the actual estimated slope. If we estimate it is less than 1, it's probably not a performance issue.